### PR TITLE
BUG: Fix test_ctypeslib and test_indexing for debug interpreter

### PR DESCRIPTION
**Pull request by tadeu**
…preter

See #7563

### CODE REVIEW

File: numpy/tests/test_ctypeslib.py [hunk 1]
```diff
@@ -8,7 +8,14 @@
 from numpy.testing import TestCase, run_module_suite, dec
 
 try:
-    cdll = load_library('multiarray', np.core.multiarray.__file__)
+    cdll = None
+    if hasattr(sys, 'gettotalrefcount'):
+        try:
+            cdll = load_library('multiarray_d', np.core.multiarray.__file__)
+        except OSError:
```
Comment (charris) [inline numpy/tests/test_ctypeslib.py]: I suspect the fix should be in `load_library`, which already tries two variant file name extensions from `get_shared_lib_extension`. Remind me of where exactly the `_d` comes from?

**Comment by tadeu** [comment]
The `_d` suffix is expected for `pyd` files when using the debug version of `python` (`python_d.exe`).

There was a previous PR doing the change in `load_library` (PR #7563), but it was rejected, and in the discussion it was decided that the fix should be in the client side:

> Yes, afaics the _d suffix needs to be added on the user side, not in load_library. (@pv)

https://github.com/numpy/numpy/pull/7563#issuecomment-212128104

I think it is safer to fix in the test too, since `load_library` is generic for any `dll`, and not specific to `pyd` files.

File: numpy/core/tests/test_indexing.py [hunk 2]
```diff
@@ -15,7 +15,14 @@
 
 
 try:
-    cdll = np.ctypeslib.load_library('multiarray', np.core.multiarray.__file__)
+    cdll = None
+    if hasattr(sys, 'gettotalrefcount'):
```
Comment (charris) [inline numpy/core/tests/test_indexing.py]: A note that this is a check for running in a debug version of Python would be helpful.

**Comment by charris** [comment]
Thanks @tadeu .

**Pull request merged**